src/main/java/appeng/services/CompassService.java

```diff
@@ -71,6 +75,22 @@ public CompassService( @Nonnull final File worldCompassFolder, @Nonnull final Th
 		return this.executor.submit( new CMDirectionRequest( coord, maxRange, cc ) );
 	}
 
+	/**
+	 * Ensure the a compass service is removed once a world gets unloaded by forge.
+	 *
+	 * @param event the event containing the unloaded world.
+	 */
+	@SubscribeEvent
+	public void unloadWorld( WorldEvent.Unload event )
+	{
+		if( Platform.isServer() )
+		{
+			final CompassReader compassReader = this.worldSet.remove( event.world );
```

Comment:
Good point. There should be a small chance that the world is not contained. I will update it.

There is probably also a chance that `event.world` might be null. But that is not an issue for a `HashMap` and otherwise when sending an event with `null` as world, they deserve a crash. :grinning: